Tools/MaterialConverter/material_converter.cpp

```diff
@@ -0,0 +1,498 @@
+        new_texture->SetName(old_input_value.tex_value->GetName());
+        input_map = Baikal::InputMap_Sampler::Create(new_texture);
+        if (input_name == "albedo")
+        {
+            // Convert texture from gamma to linear space
+            input_map = Baikal::InputMap_Pow::Create(input_map, kGammaPower);
+        }
+    }
+        break;
+    default:
+        throw std::runtime_error("Translation of unsupported input");
+    }
+    return input_map;
+}
+
+Baikal::UberV2Material::Ptr MaterialConverter::TranslateSingleBxdfMaterial(BaikalOld::SingleBxdf::Ptr old_mtl)
+{
+    LOG("Translating SingleBxdf material: " << old_mtl->GetName());
+    auto uber_mtl = Baikal::UberV2Material::Create();
+    uber_mtl->SetName(old_mtl->GetName());
+    switch (old_mtl->GetBxdfType())
+    {
+    case BaikalOld::SingleBxdf::BxdfType::kZero:
+        LOG("BxdfType: Zero");
+        uber_mtl->SetLayers(Baikal::UberV2Material::kDiffuseLayer);
+        // Black diffuse
+        uber_mtl->SetInputValue("uberv2.diffuse.color", Baikal::InputMap_ConstantFloat3::Create(RadeonRays::float3(0.0f, 0.0f, 0.0f)));
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kLambert:
+        LOG("BxdfType: Lambert");
+        uber_mtl->SetLayers(Baikal::UberV2Material::kDiffuseLayer);
+        uber_mtl->SetInputValue("uberv2.diffuse.color", TranslateInput(old_mtl, "albedo"));
+        if (old_mtl->GetInputValue("normal").tex_value != nullptr)
+        {
+            LOG("Found normal");
+            uber_mtl->SetLayers(Baikal::UberV2Material::kDiffuseLayer | Baikal::UberV2Material::kShadingNormalLayer);
+            uber_mtl->SetInputValue("uberv2.shading_normal", Baikal::InputMap_Remap::Create(
+                Baikal::InputMap_ConstantFloat3::Create(RadeonRays::float3(0.0f, 1.0f, 0.0f)),
+                Baikal::InputMap_ConstantFloat3::Create(RadeonRays::float3(-1.0f, 1.0f, 0.0f)),
+                TranslateInput(old_mtl, "normal")));
+        }
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kIdealReflect:
+    case BaikalOld::SingleBxdf::BxdfType::kMicrofacetBeckmann:
+    case BaikalOld::SingleBxdf::BxdfType::kMicrofacetGGX:
+        LOG("BxdfType: Reflection");
+        uber_mtl->SetLayers(Baikal::UberV2Material::kReflectionLayer);
+        uber_mtl->SetInputValue("uberv2.reflection.color", TranslateInput(old_mtl, "albedo"));
+        uber_mtl->SetInputValue("uberv2.reflection.roughness", TranslateInput(old_mtl, "roughness"));
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kEmissive:
+        LOG("BxdfType: Emission");
+        uber_mtl->SetLayers(Baikal::UberV2Material::kEmissionLayer);
+        uber_mtl->SetInputValue("uberv2.emission.color", TranslateInput(old_mtl, "albedo"));
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kPassthrough:
+        LOG("BxdfType: Passthrough");
+        // Nothing?
+        break;
+    case BaikalOld::SingleBxdf::BxdfType::kTranslucent:
+        LOG("BxdfType: Translucensy");
```

Comment:
Add assertion please. UberV2 don't support translucency. We got task to implement it.